### PR TITLE
Py tensor star fixes

### PR DESCRIPTION
**Pull request by blair**
###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 10.14.6 18G95
Xcode 10.3 10G8 

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by mf2k** [comment]
Error is:

```
  File "setup.py", line 20, in <module>
    from setuptools import setup, find_packages
ModuleNotFoundError: No module named 'setuptools'
```

It appears that it needs a build dependency on py-setuptools.

**Comment by cjones051073** [comment]
Please do not just go on the pypi metadata to define dependencies. I would like to see the configure step with and without the ones you have removed to be sure they really are not needed. I dont think I would have added them in the first place for just no reason...

**Comment by blair** [comment]
For py-tensorflow the dependencies I removed:

* port:py${python.version}-future
    * Looking into this I discovered the confusing `future` and `futures` ports. `futures` (with the s) provides only for py27 backport of the `concurrent.futures. `future` (no s) is for converting code between Python 2 and 3.
    * `port contents py37-tensorflow| grep -v '.pyc' | xargs grep import | grep future | grep -v __future__` doesn't show any matches. Also, there are no matches of `futurize` or `pasteurize`
    * Do you remember the reason for adding it on June 22nd? Or should have that been `port:py${python.version}-futures`. 

* port:py${python.version}-mock
    * The bug mentioned in the Portfile when the dependency on py-mock was added to Portfile references https://github.com/tensorflow/tensorflow/issues/25282 . The bug was fixed upstream by updating the absl dependency. In the mean time, I updated py-absl to the latest 0.7.1. So with py-absl this works for me in a python 3.7 shell: `from tensorflow.python.platform.googletest import mock`. However, if I go back to the previous py-absl 0.2.0 that MacPorts had then `from tensorflow.python.platform.googletest import mock` does error out. So maybe it works around the issue? In any case, I need to remove the mock reinplace code in the Portfile.

For py-tensorboard:

* port:py${python.version}-bleach
    * The py-tensorboard has it's own private copy of bleach: `from tensorboard._vendor import bleach`

* port:py${python.version}-html5lib
    * Same thing, it has it's own private copy of html5lib: `from tensorboard._vendor import html5lib`

I cannot find any references to bleach and html5lib outside the vendor version. bleach and html5lib were there in the original creation of the port in April 5th, 2018 for version 1.6.0. So there appears to be no point in pulling in MacPorts ports for these.

One small request, please document, e.g. for the other committers, why a dependency was added in the commit so future developers know if and why it's there. Not having any documentation makes it harder to determine if something can be safely removed.

Let me know if you'd like any other changes. We can wait for the builds to complete to see how they do.

**Comment by blair** [comment]
BTW, thanks for asking me to review the dependencies, I think the ports will be in better shape after this.

File: python/py-tensorboard/Portfile [hunk 1]
```diff
@@ -74,8 +74,4 @@ if {${name} ne ${subport}} {
         --root ${destroot} \
         ${distpath}/${distfiles}
     destroot.post_args
-
-    livecheck.type      none
```
Comment (cjones051073) [inline python/py-tensorboard/Portfile]: Please put this back. It is intentional that the version specific ports have livecheck none.

File: python/py-tensorflow/Portfile [hunk 2]
```diff
@@ -189,12 +196,5 @@ if {${name} ne ${subport}} {
     post-destroot {
         file delete ${destroot}${python.prefix}/bin/tensorboard
         file delete ${destroot}${prefix}/bin/tensorboard-${python.branch}
-        # https://github.com/tensorflow/tensorflow/issues/25282
-        foreach f [ exec find ${destroot}${python.prefix} -name "__init__.py" ] {
-            reinplace -q "s|from tensorflow.python.platform.googletest import mock|import mock|g" ${f}
-        }
     }
-
-    livecheck.type  none
```
Comment (cjones051073) [inline python/py-tensorflow/Portfile]: Please out this back.

**Comment by cjones051073** [comment]
Thanks for reviewing the dependencies. I am afraid I cannot recall the exact reasons for adding specific dependencies. What you propose above seems reasonable enough.

File: python/py-tensorboard/Portfile (hunk 1 above)
Comment (blair) [reply python/py-tensorboard/Portfile]: I don't agree with this, but I'll back it out. We can decide as a group on macports-dev@.

Comment (cjones051073) [reply python/py-tensorboard/Portfile]: I am find with a open discussion on this, but yes for now please out them back. I for one like the current behaviour as it works well with how I personally check the ports I maintain for updates.

**Comment by blair** [comment]
@cjones051073 Does this look good to submit without the livecheck changes? The builds are timing out, the longest one at 6+ hours?

**Comment by cjones051073** [comment]
> @cjones051073 Does this look good to submit without the livecheck changes? The builds are timing out, the longest one at 6+ hours?

I guess so yes. Builds are probably always going to fail, as each does not a long time.

**Pull request merged**